packages/core/src/renderer/dragUtils.ts

```diff
@@ -12,42 +13,50 @@ export const dragUpdate = (
   dy: number
 ): State => {
   const xs = [...state.varyingValues];
+  const frozenVaryingValues = [];
   for (const shape of state.shapes) {
     if (shape.properties.name.contents === id) {
-      dragShape(shape, [dx, dy], xs);
+      const ids = dragShape(shape, [dx, dy], xs);
+      frozenVaryingValues.push(...ids);
     }
   }
+  const { inputs, constrFns, objFns } = state;
   const updated: State = {
     ...state,
-    params: { ...state.params, optStatus: "NewIter" },
+    params: genOptProblem(
+      inputs,
+      objFns.map(({ output }) => output),
+      constrFns.map(({ output }) => output),
+      frozenVaryingValues
+    ),
```

Comment:
I'd rather not call `genOptProblem` on every drag, because that calls `makeGraph` and `genCode`, both of which are expensive. Instead of doing this, could you pass `frozenValues` from the state to the optimizer when you run it? There shouldn't be any need to recompile, because the computation graph itself remains exactly the same after a drag.